src/pip/_internal/req/req_install.py

```diff
@@ -862,17 +862,24 @@ def install(
 
         self.install_succeeded = success
 
-        if success and self.legacy_install_reason == 8368:
-            deprecated(
-                reason=(
-                    "{} was installed using the legacy 'setup.py install' "
-                    "method, because a wheel could not be built for it.".
-                    format(self.name)
-                ),
-                replacement="to fix the wheel build issue reported above",
-                gone_in="21.0",
-                issue=8368,
-            )
+        if success:
+            if self.legacy_install_reason == 8368:
+                deprecated(
+                    reason=(
+                        "{} was installed using the legacy 'setup.py install' "
+                        "method, because a wheel could not be built for it.".
+                        format(self.name)
+                    ),
+                    replacement="to fix the wheel build issue reported above",
+                    gone_in="21.0",
+                    issue=8368,
+                )
+            if self.link.is_vcs:
+                logger.warning(
+                    "direct URL of package %s will not be recorded when "
```

Comment:
```suggestion
                    "Direct URL of package %s will not be recorded when "
```

It may be a good place to sugget the user needs to install 'wheel' as a fix.

Reply:
done, see now